### PR TITLE
Fix Typos

### PR DESCRIPTION
**Pull request by Coeur**
CLA: trivial

##### Checklist
- [x] documentation is added or updated

### CODE REVIEW

File: crypto/lhash/lhash.c [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
 /*
  * A hashing implementation that appears to be based on the linear hashing
- * alogrithm:
+ * algorithm:
  * https://en.wikipedia.org/wiki/Linear_hashing
  *
  * Litwin, Witold (1980), "Linear hashing: A new tool for file and table
  * addressing", Proc. 6th Conference on Very Large Databases: 212-223
- * http://hackthology.com/pdfs/Litwin-1980-Linear_Hashing.pdf
+ * https://hackthology.com/pdfs/Litwin-1980-Linear_Hashing.pdf
```
Comment (mspncp) [inline crypto/lhash/lhash.c]: Not a typo and the old link still works. But a good change nevertheless.

File: demos/bio/descrip.mms [hunk 2]
```diff
@@ -37,7 +37,7 @@ server-arg.exe : server-arg.obj
 server-cmod.exe : server-cmod.obj
 server-conf.exe : server-conf.obj
 
-# Stoopid MMS doesn't infer this automatically...
+# MMS doesn't infer this automatically...
```
Comment (mspncp) [inline demos/bio/descrip.mms]: LOL

**Comment by Coeur** [comment]
I've added a commit for a final minor typo that I just found: you can squash the two commits when done reviewing.
Yes, I do run a spellchecker: the one from AppCode (https://www.jetbrains.com/objc/).

**Comment by mspncp** [comment]
> you can squash the two commits when done reviewing.

Sure, I would have squashed them anyway :-). And thanks for the link.

**Comment by mspncp** [comment]
You can just use `git commit --fixup` if you are used to it. We'll handle that properly.

File: crypto/lhash/lhash.c (hunk 1 above)
Comment (Coeur) [reply crypto/lhash/lhash.c]: I didn't search for all the http: schemes, but I stumble upon this one while fixing the surrounding spelling issues.

File: demos/bio/descrip.mms (hunk 2 above)
Comment (levitte) [reply demos/bio/descrip.mms]: This is not a typo! Just sayin' :wink:

**Comment by mspncp** [comment]
> I agree re trivial

Me too.

**Comment by mspncp** [comment]
Merged to master.

But the _cherry-pick to 1.1.1 fails with a lot of conflicts_. Probably it would make more sense to run the spell checker on the 1.1.1 branch and submit a separate pull request for 1.1.1. What do you think @Coeur?


```
msp@msppc:~/openssl/openssl$ git status
On branch OpenSSL_1_1_0-stable
Your branch is up to date with 'origin/OpenSSL_1_1_0-stable'.

You are currently cherry-picking commit c2969ff6e7.
  (fix conflicts and run "git cherry-pick --continue")
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:

	modified:   crypto/bio/bss_dgram.c
	modified:   crypto/bn/bn_lcl.h
	modified:   crypto/dsa/dsa_ossl.c
	modified:   crypto/ec/ec_lcl.h
	modified:   crypto/lhash/lhash.c
	modified:   crypto/sha/asm/sha512-sparcv9.pl
	modified:   demos/bio/descrip.mms
	modified:   demos/evp/aesgcm.c
	modified:   util/perl/OpenSSL/Test.pm

Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)

	both modified:   CHANGES
	deleted by us:   Configurations/15-ios.conf
	deleted by us:   Configurations/50-win-onecore.conf
	deleted by us:   Configurations/common0.tmpl
	both modified:   apps/pkcs12.c
	both modified:   apps/speed.c
	deleted by us:   apps/storeutl.c
	deleted by us:   crypto/README.sparse_array
	both modified:   crypto/aes/asm/aes-s390x.pl
	both modified:   crypto/asn1/a_time.c
	both modified:   crypto/bn/asm/mips.pl
	both modified:   crypto/bn/bn_div.c
	both modified:   crypto/ec/asm/ecp_nistz256-x86_64.pl
	deleted by us:   crypto/ec/asm/x25519-ppc64.pl
	both modified:   crypto/ec/ecp_nistp521.c
	both modified:   crypto/ec/ecx_meth.c
	deleted by us:   crypto/err/openssl.txt
	both modified:   crypto/evp/evp_err.c
	deleted by us:   crypto/initthread.c
	both modified:   crypto/poly1305/asm/poly1305-ppc.pl
	deleted by us:   crypto/rand/drbg_lib.c
	both modified:   crypto/rand/rand_lib.c
	both modified:   crypto/rsa/rsa_gen.c
	deleted by us:   crypto/sm2/sm2_sign.c
	deleted by us:   crypto/sparse_array.c
	deleted by us:   crypto/store/loader_file.c
	both modified:   doc/crypto/BIO_s_mem.pod
	both modified:   doc/crypto/EVP_DigestSignInit.pod
	both modified:   doc/crypto/EVP_DigestVerifyInit.pod
	deleted by us:   doc/internal/man3/openssl_ctx_get_data.pod
	deleted by us:   doc/internal/man3/ossl_method_construct.pod
	deleted by us:   doc/internal/man3/ossl_provider_new.pod
	deleted by us:   doc/man1/mac.pod
	deleted by us:   doc/man3/ADMISSIONS.pod
	deleted by us:   doc/man3/EVP_MD_fetch.pod
	deleted by us:   doc/man3/EVP_md5.pod
	deleted by us:   doc/man3/OSSL_CRMF_pbmp_new.pod
	deleted by us:   doc/man3/OSSL_PARAM_int.pod
	deleted by us:   doc/man3/OSSL_STORE_LOADER.pod
	deleted by us:   doc/man3/OSSL_STORE_expect.pod
	deleted by us:   doc/man3/RAND_DRBG_set_callbacks.pod
	deleted by us:   doc/man3/SSL_CTX_set_cipher_list.pod
	deleted by us:   doc/man3/SSL_CTX_set_srp_password.pod
	deleted by us:   doc/man3/SSL_SESSION_get0_hostname.pod
	deleted by us:   doc/man3/SSL_write.pod
	deleted by us:   doc/man7/EVP_KDF_SS.pod
	deleted by us:   doc/man7/EVP_KDF_SSHKDF.pod
	deleted by us:   doc/man7/EVP_KDF_X963.pod
	deleted by us:   doc/man7/property.pod
	deleted by us:   engines/e_devcrypto.c
	deleted by us:   include/internal/property.h
	both modified:   include/internal/thread_once.h
	deleted by us:   include/internal/tsan_assist.h
	deleted by us:   test/asn1_time_test.c
	deleted by us:   test/drbgtest.c
	both modified:   test/dtlstest.c
	both modified:   test/evp_extra_test.c
	deleted by us:   test/params_test.c
	both modified:   test/ssltestlib.c
	deleted by us:   test/tls13secretstest.c
```

**Comment by mspncp** [comment]
There are some things about `git status` which I don't understand. For example, why does it report the following file as deleted, although it exists in master and 1.1.1?

```
deleted by us:   doc/man3/RAND_DRBG_set_callbacks.pod
```

**Comment by mspncp** [comment]
Oh, I'm stupid: it's not the files which were deleted, it's the changes! Probably this simply means that the typos are not present in 1.1.1 (yet) and the `deleted by us` conflicts can be ignored when merging.

@Coeur would you mind doing the merge to 1.1.1 yourself (or running an independent spell check on 1.1.1, as you prefer) and create a separate pull request for it?

**Comment by Coeur** [comment]
I'm aware of the many differences between 1.1.1 and master. I can make a separate PR for 1.1.1: but this current PR was for master only, so you can close it once merged.

**Comment by mspncp** [comment]
Ok, then I'll close it now. Thanks again!

**Pull request closed without merging**